### DISCUSSION TITLE
Cubic Signed Distance Fields

### DISCUSSION THREAD

**Discussion by jost0** [InteractiveComputerGraphics/PositionBasedDynamics#154] [General]
Hi, I'm trying to create a scene with cubic signed distance field loaded from a file generated by discregrid.

For simplicity, I use an already generated file "/data/sdf/bunny_10k.csdf" provided by the author.

My problem is that when I try and build the program I get a linker error:
 error LNK2019: unresolved external symbol "public: __cdecl Discregrid::CubicLagrangeDiscreteGrid::CubicLagrangeDiscreteGrid(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)"

This might originate from an erroneous way of loading the SDF.

See below for the code where I load the SDF and call the addCubicSDFCollisionObject.

`

        string fileNameSDF = FileSystem::normalizePath(base->getExePath() + "/data/sdf/bunny_10k.csdf");

	string fileNameObj = FileSystem::normalizePath(base->getExePath() + "/resources/models/bunny_10k.obj");

	IndexedFaceMesh meshBunny;
	VertexData vdBunny;

	DemoBase::loadMesh(fileNameObj, vdBunny, meshBunny, Vector3r::Zero(), Matrix3r::Identity(), Vector3r::Ones());
	meshBunny.setFlatShading(true);

	rb[rbIndex] = new RigidBody();
	rb[rbIndex]->initBody(100, Vector3r(0.0, 0.0, 0.0), Quaternionr(0, 0, 0, 0), vdBunny, meshBunny, Vector3r::Ones());

	const std::vector<Vector3r>& verticesBunny = rb[rbIndex]->getGeometry().getVertexDataLocal().getVertices();

	const unsigned int numVert = static_cast<unsigned int>(verticesBunny.size());
	unsigned int bodyIndex = 0;
	const unsigned int test = 5;
	const unsigned int tester = rbIndex;
	cSDF->addCubicSDFCollisionObject(tester, test, verticesBunny.data(), numVert, fileNameSDF, Vector3r::Ones(), true, false);`

So my question in this endeavour is, am I doing this properly? Is there already a demo which contains the useage of cubic SDFs for collision detection?

Thanks in advance.

**Comment by janbender** [comment]
Hi, 

First, the Quaternion in your example "Quaternionr(0, 0, 0, 0)" does not represent a rotation. Try to use "Quaternionr(1.0, 0.0, 0.0, 0.0)" which represents a rotation by 0 degree. 

Don't know if this solves your problem. If not, please take a look at this example here:

https://github.com/InteractiveComputerGraphics/PositionBasedDynamics/discussions/147#discussioncomment-4231696

I hope this helps.

**Reply by jost0** [reply]
Thank you for the quick reply!
I will look into #147, which is exactly what I am after!
That helps massively, thank you again!
